lib/analyzer/image-inspector.ts

```diff
@@ -12,6 +18,83 @@ async function detect(
   return JSON.parse(info.stdout)[0];
 }
 
+/**
+ * TODO: This function could probably have a better name and location
+ *
+ * In the case that an `ImageType.Identifier` is detected we need to produce
+ * an image archive, either by saving the image if it's already loaded into
+ * the local docker daemon, or by pulling the image from a remote registry and
+ * saving it to the filesystem directly.
+ *
+ * Users may also provide us with a URL to an image in a Docker compatible
+ * remote registry.
+ *
+ * @param {string} targetImage
+ * @param {string} [username]
+ * @param {string} [password]
+ */
+async function getArchiveLocation(
+  targetImage: string,
+  username?: string,
+  password?: string,
+): Promise<string> {
+  const docker = new Docker(targetImage);
+
+  try {
```

Comment:
what is the purpose of this try-catch? to use the docker binary if it exists?

Reply:
No, it's checking if the image is available from the local docker daemon before we reach out to the remote registry to download it.